src/region.rs

```diff
@@ -59,3 +61,28 @@ impl From<u8> for Region {
         Region::from(x as usize)
     }
 }
+
+impl FromStr for Region {
+    type Err = Error;
+    fn from_str(x: &str) -> Result<Self, Self::Err> {
+        use Region::*;
+        match x {
```

Comment:
If you Ok-wrap the entire match you can save yourself a lot of typing Ok every branch